tfx/components/base/base_executor_test.py

```diff
@@ -37,21 +44,130 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+      beam_inputs = self.read_inputs(p, input_dict, output_dict,
+                                     exec_properties)
+      beam_outputs = self.construct_beam_graph(p, beam_inputs, input_dict,
+                                               output_dict, exec_properties)
+      self.write_outputs(p, beam_outputs, input_dict, output_dict,
+                         exec_properties)
+
 class BaseExecutorTest(tf.test.TestCase):
 
   def testBeamSettings(self):
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--runner=DirectRunner'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(StandardOptions)
+    options = executor._make_beam_pipeline().options.view_as(StandardOptions) # pylint: disable=protected-access
     self.assertEqual('DirectRunner', options.runner)
 
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--direct_num_workers=2'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(DirectOptions)
+    options = executor._make_beam_pipeline().options.view_as(DirectOptions) # pylint: disable=protected-access
     self.assertEqual(2, options.direct_num_workers)
 
 
+class BeamExecutorTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(BeamExecutorTest, self).setUp()
+    self._input_data_path = (
+        os.path.join(os.environ.get(
+            'TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()), 'input.txt'))
+    output_data_path = (
+        os.path.join(os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR',
+                                    self.get_temp_dir()), 'output.txt'))
+    self._full_output_data_path = output_data_path + '-00000-of-00001'
+
+    self._generate_data(self._input_data_path)
+
+    # Create input dict
+    examples = types.standard_artifacts.Examples()
+    examples.uri = self._input_data_path
+    self._input_dict = {_EXAMPLES: examples}
+
+    # Create output dict
+    shuffled_examples = types.standard_artifacts.Examples()
+    shuffled_examples.uri = output_data_path
+    self._output_dict = {_SHUFFLED_EXAMPLES: shuffled_examples}
+
+    # Run Do() method
+    beam_test_executor = _BeamTestExecutor()
+    beam_test_executor.Do(self._input_dict, self._output_dict, {})
+
+  def _generate_data(self, file_path):
+    with open(file_path, "w+") as f:
+      for i in range(_NUM_LINES):
+        line = str(i) + '\n'
+        f.write(line)
+
+  def testBeamExecutorBeamIoSignature(self):
+    beam_test_executor = _BeamTestExecutor()
+    input_signature, output_signature = (
+        beam_test_executor.beam_io_signature(
+            self._input_dict, self._output_dict, {}))
+
+    with open(self._input_data_path) as f:
+      self.assertTrue(isinstance(f, input_signature[_EXAMPLES]))
+
+    with open(self._full_output_data_path) as f:
+      self.assertTrue(isinstance(f, output_signature[_SHUFFLED_EXAMPLES]))
```

Comment:
Similarly, can you assert that `self._output_dict[_EXAMPLES]` has the appropriate output signature (i.e. `int`)?  Note that the system should be able to infer this (otherwise, you would need to use `.with_output_types(int)` during the `construct_beam_graph()` method).

Reply:
I got rid of the file openings, and assert that `input_signature[_EXAMPLES] = Text` and `output_signature[_SHUFFLED_EXAMPLES] = int`. I'm unable to force `.with_output_types(int)` in `construct_beam_graph()` since I get the error `  File "apache_beam/coders/stream.pyx", line 225, in apache_beam.coders.stream.get_varint_size
TypeError: an integer is required [while running 'WriteToText/Write/WriteImpl/FinalizeWrite']`, so I have to do `actual_sum = int(f.read())` in `def testFuseableBeamExecutorDo(self):`. Let me know if this is the correct approach.